internal/engine/ingester/diff/diff.go

```diff
@@ -203,6 +208,124 @@ func (di *Diff) ingestFileForDepDiff(
 	return batchCtxDeps, nil
 }
 
+func (di *Diff) getScalibrTypeDiff(ctx context.Context, _ int, pr *pb.PullRequest) (*engif.Result, error) {
+	deps := pbinternal.PrDependencies{Pr: pr}
+
+	// TODO: we should be able to just fetch the additional commits between base and target.
+	// Our current Git abstraction isn't quite powerful enough, so we do two shallow clones.
+
+	baseInventory, err := di.scalibrInventory(ctx, pr.BaseCloneUrl, pr.BaseRef)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to clone base from %s at %q: %w", pr.BaseCloneUrl, pr.BaseRef, err)
+	}
+	newInventory, err := di.scalibrInventory(ctx, pr.TargetCloneUrl, pr.TargetRef)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to clone fork from %s at %q: %w", pr.TargetCloneUrl, pr.TargetRef, err)
+	}
+
+	newDeps := setDifference(baseInventory, newInventory, inventorySorter)
+
+	deps.Deps = make([]*pbinternal.PrDependencies_ContextualDependency, 0, len(newDeps))
+	for _, inventory := range newDeps {
+		for _, filename := range inventory.Locations {
+			deps.Deps = append(deps.Deps, &pbinternal.PrDependencies_ContextualDependency{
+				Dep: &pbinternal.Dependency{
+					Ecosystem: inventoryToEcosystem(inventory),
+					Name:      inventory.Name,
+					Version:   inventory.Version,
+				},
+				File: &pbinternal.PrDependencies_ContextualDependency_FilePatch{
+					Name:     filename,
+					PatchUrl: "", // TODO: do we need this?
+				},
+			})
+		}
+	}
+
+	return &engif.Result{Object: &deps, Checkpoint: checkpoints.NewCheckpointV1Now()}, nil
+}
+
+func inventorySorter(a *extractor.Inventory, b *extractor.Inventory) int {
+	return cmp.Or(cmp.Compare(a.Name, b.Name), cmp.Compare(a.Version, b.Version))
```

Comment:
Can you  get packages whose name and version are the same? (e.g. two identical lockfiles scanned from different files)
We have an existing comparison function for Inventory in SCALIBR that looks at all the relevant fields of the inventory: https://github.com/google/osv-scalibr/blob/main/scalibr.go#L293
If you think it'd beneficial to use for you we can make it public.